modules/internal/LastResortIO.chpl

```diff
@@ -0,0 +1,389 @@
+  return stringStyleTerminated(terminator);
+}
+*/
+pragma "last resort"
+proc stringStyleNullTerminated() {
+  compilerWarning("stringStyleNullTerminated will no longer be available by default,\nplease insert a use of the IO module to continue calling it.");
+  return stringStyleNullTerminated();
+}
+
+pragma "last resort"
+proc stringStyleExactLen(len:int(64)) {
+  compilerWarning("stringStyleExactLen will no longer be available by default,\nplease insert a use of the IO module to continue calling it.");
+  return stringStyleExactLen(len);
+}
+
+pragma "last resort"
+proc stringStyleWithVariableLength() {
+  compilerWarning("stringStyleWithVariableLength will no longer be available by default,\nplease insert a use of the IO module to continue calling it.");
+  return stringStyleWithVariableLength();
+}
+
+pragma "last resort"
+proc stringStyleWithLength(lengthBytes:int) throws {
+  compilerWarning("stringStyleWithLength will no longer be available by default,\nplease insert a use of the IO module to continue calling it.");
+  return stringStyleWithLength(lengthBytes);
+}
+
+pragma "last resort"
+proc IOHINT_NONE {
+  compilerWarning("IOHINT_NONE will no longer be available by default, please\ninsert a use of the IO module to access it.");
+  return IO.IOHINT_NONE;
+}
+
+pragma "last resort"
+proc IOHINT_RANDOM {
+  compilerWarning("IOHINT_RANDOM will no longer be available by default, please\ninsert a use of the IO module to access it.");
+  return IO.IOHINT_RANDOM;
+}
+
+pragma "last resort"
+proc IOHINT_SEQUENTIAL {
+  compilerWarning("IOHINT_SEQUENTIAL will no longer be available by default, please\ninsert a use of the IO module to access it.");
+  return IO.IOHINT_SEQUENTIAL;
+}
+
+pragma "last resort"
+proc IOHINT_CACHED {
+  compilerWarning("IOHINT_CACHED will no longer be available by default, please\ninsert a use of the IO module to access it.");
+  return IO.IOHINT_CACHED;
+}
+
+pragma "last resort"
+proc IOHINT_PARALLEL {
+  compilerWarning("IOHINT_PARALLEL will no longer be available by default, please\ninsert a use of the IO module to access it.");
+  return IO.IOHINT_PARALLEL;
+}
+
+// TODO: iostyle record
+/*pragma "last resort"
+proc iostyle type {
+  compilerWarning("iostyle will no longer be available by default, please\ninsert a use of the IO module to access it.");
+  return IO.iostyle;
+  }*/
+
+pragma "last resort"
+proc defaultIOStyle() {
+  compilerWarning("defaultIOStyle will no longer be available by default,\nplease insert a use of the IO module to continue calling it.");
+  return defaultIOStyle();
+}
+
+pragma "last resort"
+proc iohints type {
+  compilerWarning("iohints will no longer be available by default, please\ninsert a use of the IO module to access it.");
+  return IO.iohints;
+}
+
+// TODO: file record methods
+pragma "last resort"
+proc file type {
+  compilerWarning("file will no longer be available by default, please\ninsert a use of the IO module to access it.");
+  return IO.file;
+}
```

Comment:
I was also expecting a `channel` type function here?

I feel pretty good about having wrappers only for:
 * types channel and file
 * non-method functions (free functions) - especially `open` and the like

Reply:
The only reason I didn't was because I got tired and didn't want to hold up the PR any further.  I absolutely intend to get to doing that soon, but was hoping to merge this sooner so that we could see the compiler performance benefit (which I am now skeptical will happen, since I think the deprecation warnings are undoing the benefit not having the symbols caused, but that's another issue...)

Reply:
Right, but it seems to me to be sufficient to focus on the common types and free functions for the warning. That would address the problem of un-doing performance benefit, wouldn't it?

Reply:
Looks like `channel` causes problems where `file` does not.  I think this is because ChapelIO needs channel for the functions and methods it contains and uses both LastResortIO and IO.  I can try making the use of LastResortIO slightly deeper (so inserting a fake module which uses it, and having ChapelIO use that fake module) but I would rather not place the use in someplace more prominent like ChapelStandard, for instance, since I think that would open a can of worms.

Reply:
Huh, I would think that the "last resort" flag would handle it. Maybe "last resort" doesn't work for paren-less functions for some reason?

Reply:
We don't check against the "last resort" flag during scope resolution, which is when we're getting the error.  Perhaps we should?  Part of the point of that error message is to prevent the abuse I'm doing, it'd really be better for us to have a way to mark symbols other than functions as deprecated, as #14396 brings up